File: _courses/spring-data-jpa-book/src/main/java/io/beaniejoy/springdatajpabook/controller/MemberController.java
package io.beaniejoy.springdatajpabook.controller;

import io.beaniejoy.springdatajpabook.dto.MemberResponse;
import io.beaniejoy.springdatajpabook.entity.Member;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class MemberController {

    @GetMapping("/member")
    public MemberResponse findMember(@RequestParam("id") Member member) {
        return toResponse(member);
    }

    private MemberResponse toResponse(Member member) {
        if (member == null) return null;
        return new MemberResponse(member.getId(), member.getName(), member.getEmail());
    }
}

File: spring-test-fixtures/application/src/main/kotlin/io/beaniejoy/springtestfixtures/controller/MemberController.kt
package io.beaniejoy.springtestfixtures.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/member")
class MemberController {
    @GetMapping
    fun list() {

    }
}